src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs

```diff
@@ -193,6 +193,76 @@ public void PublishWithRuntimeIdentifier(bool publishNoBuild)
             }
         }
 
+        [WindowsOnlyTheory]
+        [InlineData("net7.0", "win-x64", "win-x86", false, false)]
+        [InlineData("net7.0", "win-x64", "win-x86", true, false)]
+        [InlineData("net7.0", "win-x64", "win-x86", true, true)]
+        public void PublishRuntimeIdentifierSetsRuntimeIdentifierAndDoesOrDoesntOverrideRID(string tfm, string publishRuntimeIdentifier, string runtimeIdentifier, bool runtimeIdentifierIsGlobal, bool publishRuntimeIdentifierIsGlobal)
+        {
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = tfm
+            };
+            testProject.AdditionalProperties["RuntimeIdentifier"] = runtimeIdentifier;
+            testProject.AdditionalProperties["PublishRuntimeIdentifier"] = publishRuntimeIdentifier;
+            testProject.RecordProperties("RuntimeIdentifier");
+
+            List<string> args = new List<string>
+            {
+                $"/p:_IsPublishing=true", // Normally this would be set by the CLI (OR VS Soon TM), but this calls directly into t:/Publish.
+                runtimeIdentifierIsGlobal ? $"/p:RuntimeIdentifier={runtimeIdentifier}" : "",
+                publishRuntimeIdentifierIsGlobal ? $"/p:PublishRuntimeIdentifier={publishRuntimeIdentifier}" : ""
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: $"{publishRuntimeIdentifierIsGlobal}-{runtimeIdentifierIsGlobal}");
+            var publishCommand = new PublishCommand(testAsset);
+            publishCommand
+                .Execute(args.ToArray())
+                .Should()
+                .Pass();
+
+            string expectedRid = runtimeIdentifierIsGlobal ? runtimeIdentifier : publishRuntimeIdentifier;
+            var properties = testProject.GetPropertyValues(testAsset.TestRoot, targetFramework: tfm, runtimeIdentifier: expectedRid);
```

Comment:
Similar to what we talked about on another test, if the RuntimeIdentifier that the project is built with is wrong, this won't be able to read the property value.  The test should still fail but it won't be as clear why it's failing.  It's OK to leave like this, but maybe we should change the property recording logic so that it doesn't put the property values in a RuntimeIdentifier-specific folder.

EDIT: I see you noted some of this in a comment below.

Reply:
Made that change. It's going to have to go in another PR because PublishSelfContained also relies on it.